app/models/trading_partner.rb

```diff
@@ -0,0 +1,25 @@
+class TradingPartner
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :country, :string
+
+  delegate :options, to: :class
+
+  validate :validate_country
+
+  def validate_country
+    country.in?(options.map(&:id)).tap do |valid_country|
+      errors.add(:country, :invalid_country) unless valid_country
+    end
+  end
+
+  def self.options
+    GeographicalArea.all.map do |geographical_area|
```

Comment:
If `options` is only going to be used for the form builder, its probably easier to change it to call the `#long_description` method rather than wrapping the array elements in OpenStruct's ?